build/nested-cluster/integ-test.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
```

Comment:
This looks like yet another e2e wrapper. It's already hard enough to understand all levels of wrappers we have.

Do we really need to have this? and if so, does it really have to support all these environment variables?

Reply:
This isn't a new level of wrapper, it's just a wrapper for a different system. The existing e2e shell script is not particularly clean. Long-term, I'll either refactor the e2e shell script to be more modular, or I'll move this functionality into Go.

Reply:
Makes sense to write it in Go, of course, or launch it as a separate container via the API.
I've done something similar for LinuxKit recently, please take a look: https://github.com/linuxkit/kubernetes/tree/master/pkg/kube-e2e-test.

Reply:
@errordeveloper I plan to launch the tests in another container in the same pod as the API server, but not to the dind cluster itself.

@ixdy How about "dind-test.sh"?